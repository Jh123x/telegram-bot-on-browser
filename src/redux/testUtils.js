import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { botSlice, defaultBotState } from "./botSlice";

export const generateDefaultState = () => {
  return {
    bot: defaultBotState,
  };
};

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      bot: botSlice.reducer,
    },
    preloadedState,
  });
};

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { bot: botSlice.reducer },
      preloadedState: preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
