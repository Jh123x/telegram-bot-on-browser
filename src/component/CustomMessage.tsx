import React, { useState } from "react"
import { BotWithConfig, User } from "../redux/types";
import { useSelector } from "react-redux";
import { Button, Input, MenuItem, Select, Typography } from "@mui/material";
import { BrowserBot } from "../interfaces/bot";


export const CustomChat = ({ bot }: { bot?: BrowserBot }) => {
    const users = useSelector<BotWithConfig, User[]>((state) => state.bot.users);
    const [currUserID, setCurrUserID] = useState<number>(0)
    const [message, setCurrMessage] = useState<string>("")

    return <>
        <Typography variant="h3">Send a Custom message</Typography>
        <Typography variant="body1">
            You can only send messages to users you have received messages from.
        </Typography>
        <Select
            sx={{ margin: '10px' }}
            defaultValue={1}
            value={currUserID}
            onChange={(e) => {
                const target: number = e.target.value as number
                setCurrUserID(target)
                e.preventDefault();
            }}
        >
            <MenuItem value={0} disabled={true} key="default-value">Select a value</MenuItem>
            {users.map((user) => <MenuItem value={user.UserID} key={`${user.UserID}-${user.Username}`}>{user.Username}</MenuItem>)}
        </Select>
        <Input
            value={message}
            onChange={(e) => setCurrMessage(e.target.value)}
            sx={{ margin: '10px' }}
        />
        <Button
            sx={{ margin: '10px' }}
            variant="contained"
            color="success"
            onClick={() => {
                if (currUserID === 0 || !bot) return
                const botVal = bot!
                botVal.sendMessage(currUserID, message)
                setCurrMessage("")
            }}
        >Send</Button>
    </>
}