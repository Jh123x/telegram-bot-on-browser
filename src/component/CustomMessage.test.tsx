import { test, expect } from "@jest/globals";
import React from "react";
import { generateDefaultState, renderWithProviders, setupStore } from "../redux/testUtils.tsx";
import { CustomChat } from "./CustomMessage.tsx";

test("renders custom chat interface correctly", () => {
    const store = setupStore(generateDefaultState());
    const component = renderWithProviders(<CustomChat />, { store: store });
    expect(component).toMatchSnapshot();
});
