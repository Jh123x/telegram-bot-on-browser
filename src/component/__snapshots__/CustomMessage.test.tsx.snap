// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom chat interface correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
      >
        Send a Custom message
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        You can only send messages to users you have received messages from.
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiSelect-root css-1ln7fuk-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-18jp67o-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="combobox"
          tabindex="0"
        >
          Select a value
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
          tabindex="-1"
          value="0"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1nf2c5d-MuiNotchedOutlined-root"
          >
            <span
              aria-hidden="true"
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1i753ds-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input css-1yrc8ca-MuiInputBase-input-MuiInput-input"
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-17fwciq-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Send
      </button>
    </div>
  </body>,
  "container": <div>
    <h3
      class="MuiTypography-root MuiTypography-h3 css-13msox4-MuiTypography-root"
    >
      Send a Custom message
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      You can only send messages to users you have received messages from.
    </p>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiSelect-root css-1ln7fuk-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-18jp67o-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        Select a value
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        tabindex="-1"
        value="0"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1nf2c5d-MuiNotchedOutlined-root"
        >
          <span
            aria-hidden="true"
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1i753ds-MuiInputBase-root-MuiInput-root"
    >
      <input
        class="MuiInputBase-input MuiInput-input css-1yrc8ca-MuiInputBase-input-MuiInput-input"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-17fwciq-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Send
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
